File: Hydroq/data/scripts/ConfirmDialog.lua
 ConfirmDialog = { 
   OnInit = function() 
     ConfirmDialog_beh:SubscribeForMessages(StrId("BUTTON_CLICKED")) 
   end, 
   OnMessage = function(msg) 
     if(msg:HasAction(StrId("BUTTON_CLICKED")))
	 then
	   if(msg.contextNode.tag == "yes_but")
	   then
	     CogSwitchBackToScene("none")
		 CogSwitchBackToScene("none")
	   end
	   
	   if(msg.contextNode.tag == "no_but")
	   then
	     CogSwitchBackToScene("none")
	   end
	 end
   end, 
   Update = function(delta, absolute) 
   end 
} 

ConfirmDialog_beh = Behavior() 
ConfirmDialog_beh:Register(ConfirmDialog, "ConfirmDialog")

File: Hydroq/src/ConfirmDialog.cpp
#include "ConfirmDialog.h"
#include "Soundfx.h"
#include "Msg.h"
#include "Tween.h"
#include "Facade.h"
#include "Node.h"
#include "ComponentStorage.h"
#include "Stage.h"

void ConfirmDialog::OnInit() {
	SubscribeForMessages(ACT_BUTTON_CLICKED);
}

void ConfirmDialog::OnMessage(Msg& msg) {
	if (msg.HasAction(ACT_BUTTON_CLICKED)) {
		auto sceneContext = GETCOMPONENT(Stage);
		if (msg.GetContextNode()->GetTag().compare("yes_but") == 0) {
			
			// click on yes button, switch twice (first switch closes dialog)
			sceneContext->SwitchBackToScene(TweenDirection::NONE);
			sceneContext->SwitchBackToScene(TweenDirection::NONE);
		}
		else if (msg.GetContextNode()->GetTag().compare("no_but") == 0) {
			// click on no button, close the dialog
			sceneContext->SwitchBackToScene(TweenDirection::NONE);
		}
	}
}